web/src/utils/factory-management/products.ts

```diff
@@ -65,11 +65,14 @@ export const calculateProducts = (factory: Factory, gameData: DataInterface) =>
       }
 
       // === Now we need to calculate the parts required per min to make the product ===
-      const productIngredientRatio = product.amount / recipe.products[0].perMin
+      const productIngredientRatio = product.amount / recipe.products[0].perMin // This formular should have no rounding - the decimal points are important here for the floating point calculations
```

Comment:
Yep valid point, now you've added the formatNumber stuff you did before this can probably be removed. Feel free to remove the comments from this

Reply:
Yeh - keep the precision in the code, and format the number for the user in the HTML.